Limit search results to the first 500

joshmoore opened the pull request:
In order to protected the server from unnecessary
overhead, limit the number of search results to 500.
This can be improved by adding a per-call setting
and/or an overriding server-wide configuration.

cc: @aleksandra-tarkowska

joshmoore committed:
Limit search results to the first 500

In order to protected the server from unnecessary
overhead, limit the number of search results to 500.
This can be improved by adding a per-call setting
and/or an overriding server-wide configuration.

atarkowska commented:
is that failing tests https://10.0.51.106:8443/job/OMERO-test-integration/lastCompletedBuild/testReport/OmeroPy.test.integration.test_search/TestSearch/ ?

joshmoore commented:
I don't think so. I think 113 is still unhappy from previous DB changes. Probably a good time to get us moved to a new devspace.

atarkowska commented:
Search testes were ok yesterday :) hmm I didn't have problem with DB changes, need help?

joshmoore commented:
@aleksandra-tarkowska : I take it back; I think I know what's happening. The server must have a larger number of matching annotations and the limit is getting the older ones which aren't visible to the current user. This isn't something that will impact IDR, so I will likely need to disable these permission-related search tests.

atarkowska commented:
could you fix `./components/tools/OmeroPy/test/integration/test_search.py:36:19: E221 multiple spaces before operator` as well please

joshmoore commented:
Will do.

atarkowska commented:
thx

atarkowska commented on components/server/src/ome/services/search/FullText.java:
```diff
@@ -301,6 +301,7 @@ protected void initializeQuery(FullTextQuery ftQuery) {
         ftQuery
         .setProjection(ProjectionConstants.SCORE,
                 ProjectionConstants.ID);
+        ftQuery.setMaxResults(500);
```
Do you think this could become param, that will correlate with batchSize https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/tools/OmeroPy/src/omero/gateway/__init__.py#L4132 and do proper pagination?

joshmoore replied:
Somehow I missed this comment. We can certainly try for 0.3.2.

joshmoore committed:
Fix flake8

joshmoore committed:
Skip permission-based search tests

If there are more than 500 annotations matching a given
string, potentially no results will be returned for a
particular user. This isn't an issue in a public resource.

joshmoore commented:
Pushed.

```
================================= 23 passed, 5 skipped in 265.44 seconds ==================================
```

aleksandra-tarkowska commented:
--ola150

aleksandra-tarkowska commented:
--ola150

joshmoore commented:
Merging for IDR 0.3.1

atarkowska commented:
--rebased-to #5228
